src/Microsoft.AspNetCore.Authentication.Cookies/Events/CookieValidatePrincipalContext.cs

```diff
@@ -19,55 +19,19 @@ public class CookieValidatePrincipalContext : BaseCookieContext
         /// <param name="scheme"></param>
         /// <param name="ticket">Contains the initial values for identity and extra data</param>
         /// <param name="options"></param>
-        public CookieValidatePrincipalContext(HttpContext context, AuthenticationScheme scheme, AuthenticationTicket ticket, CookieAuthenticationOptions options)
-            : base(context, scheme, options, ticket?.Properties)
+        public CookieValidatePrincipalContext(
+            HttpContext context,
+            AuthenticationScheme scheme,
+            CookieAuthenticationOptions options,
+            AuthenticationTicket ticket)
+            : base(context, scheme, options)
         {
-            if (context == null)
-            {
-                throw new ArgumentNullException(nameof(context));
-            }
-
-            if (ticket == null)
-            {
-                throw new ArgumentNullException(nameof(ticket));
-            }
-
-            if (options == null)
-            {
-                throw new ArgumentNullException(nameof(options));
-            }
-
-            Principal = ticket.Principal;
+            Ticket = ticket;
         }
 
         /// <summary>
-        /// Contains the claims principal arriving with the request. May be altered to change the 
-        /// details of the authenticated user.
-        /// </summary>
-        public ClaimsPrincipal Principal { get; private set; }
-
-        /// <summary>
         /// If true, the cookie will be renewed
         /// </summary>
         public bool ShouldRenew { get; set; }
-
-        /// <summary>
-        /// Called to replace the claims principal. The supplied principal will replace the value of the 
-        /// Principal property, which determines the identity of the authenticated request.
-        /// </summary>
-        /// <param name="principal">The <see cref="ClaimsPrincipal"/> used as the replacement</param>
-        public void ReplacePrincipal(ClaimsPrincipal principal)
-        {
```

Comment:
Likely need to bring these back, this will break Identity/cookie stamp validator